### ISSUE TITLE
Warnings during nom run dev (from Webpack)

### ISSUE THREAD

**Issue by sthagen** [secvisogram/secvisogram#10]
Maybe we should adapt the build process?

While trying to investigate the date picker problems on Safari / MacOS (optional feature, not a defined  target browser) I noticed during build:

```
(node:25202) [DEP_WEBPACK_MAIN_TEMPLATE_ASSET_PATH] DeprecationWarning: MainTemplate.hooks.assetPath is deprecated (use Compilation.hooks.assetPath instead)
```
and:

```
(node:25202) [DEP_WEBPACK_COMPILATION_ASSETS] DeprecationWarning: Compilation.assets will be frozen in future, all modifications are deprecated.
BREAKING CHANGE: No more changes should happen to Compilation.assets after sealing the Compilation.
	Do changes to assets earlier, e. g. in Compilation.hooks.processAssets.
	Make sure to select an appropriate stage from Compilation.PROCESS_ASSETS_STAGE_*.
```

Complete log:
```
# ...
> webpack serve --mode development

(node:25202) [DEP_WEBPACK_MAIN_TEMPLATE_ASSET_PATH] DeprecationWarning: MainTemplate.hooks.assetPath is deprecated (use Compilation.hooks.assetPath instead)
(Use `node --trace-deprecation ...` to show where the warning was created)
ℹ ｢wds｣: Project is running at http://localhost:8080/
ℹ ｢wds｣: webpack output is served from /
ℹ ｢wds｣: Content not from webpack is served from /somewhere/secvisogram/app
ℹ ｢wdm｣: wait until bundle finished: /
ℹ ｢wdm｣: wait until bundle finished: /
(node:25202) [DEP_WEBPACK_COMPILATION_ASSETS] DeprecationWarning: Compilation.assets will be frozen in future, all modifications are deprecated.
BREAKING CHANGE: No more changes should happen to Compilation.assets after sealing the Compilation.
	Do changes to assets earlier, e. g. in Compilation.hooks.processAssets.
	Make sure to select an appropriate stage from Compilation.PROCESS_ASSETS_STAGE_*.
ℹ ｢wdm｣: assets by path vendor/ace/*.js 8.1 MiB 245 assets
assets by path vendor/ace/snippets/*.js 288 KiB 171 assets
assets by path *.js 18.9 MiB 4 assets
assets by path *.css 11.8 MiB 4 assets
assets by path *.html 2.37 KiB
  asset view-tests-canvas.html 801 bytes [emitted]
  asset index.html 789 bytes [emitted]
  asset tests.html 557 bytes [emitted]
  asset view-tests.html 282 bytes [emitted]
assets by path vendor/first/*.js 74.5 KiB
  asset vendor/first/cvsscalc31.js 38.9 KiB [emitted] [from: vendor/first/cvsscalc31.js] [copied]
  asset vendor/first/cvsscalc30.js 35.6 KiB [emitted] [from: vendor/first/cvsscalc30.js] [copied]
asset COMMITHASH 40 bytes [emitted]
asset VERSION 6 bytes [emitted]
Entrypoint secvisogram 9.68 MiB = secvisogram.css 3.94 MiB secvisogram.js 5.74 MiB
Entrypoint view-tests 8.63 MiB = view-tests.css 3.94 MiB view-tests.js 4.68 MiB
Entrypoint view-tests-canvas 8.63 MiB = view-tests-canvas.css 3.94 MiB view-tests-canvas.js 4.68 MiB
Entrypoint tests 3.82 MiB = tests.css 4.55 KiB tests.js 3.81 MiB
orphan modules 824 KiB [orphan] 399 modules
runtime modules 8.35 KiB 27 modules
cached modules 767 bytes [cached] 1 module
modules by path ./node_modules/ 5.51 MiB (javascript) 4.99 KiB (css/mini-extract) 531 modules
modules by path ./lib/ 553 KiB (javascript) 3.94 MiB (css/mini-extract)
  modules by path ./lib/SecvisogramPage/View/ 192 KiB 50 modules
  modules by path ./lib/shared/ 338 KiB (javascript) 3.94 MiB (css/mini-extract) 15 modules
  2 modules
modules by path ./viewTests/ 13.8 KiB
  modules by path ./viewTests/shared/ 11.4 KiB 4 modules
  modules by path ./viewTests/*.js 2.41 KiB 2 modules
modules by path ./tests/ 18.3 KiB 6 modules
modules by path ./seeds/documents/*.json 385 KiB
  ./seeds/documents/valid-1.json 61.9 KiB [built] [code generated]
  ./seeds/documents/valid-2.json 323 KiB [built] [code generated]
webpack 5.27.1 compiled successfully in 167169 ms
ℹ ｢wdm｣: Compiled successfully.
```

**Comment by tschmidtb51** [comment]
@sthagen: This warning is caused by the Webpack Plugin `git-revision-webpack-plugin`. There is a fix available, but no newer version: https://github.com/pirelenito/git-revision-webpack-plugin/pull/56

Unfortunately, the fix hasn't made it yet into the release which is available at: https://www.npmjs.com/package/@hzzt/git-revision-webpack-plugin

Feel free to ping the repo owners about that...

**Comment by tschmidtb51** [comment]
The PR go accepted: https://github.com/fh345392977/git-revision-webpack-plugin/pull/1 However, as far as I know a new version hasn't been publish yet (still 3.0.7).

**Comment by tschmidtb51** [comment]
There are 2 new versions:

- https://www.npmjs.com/package/@hzzt/git-revision-webpack-plugin (3.0.8)
- https://www.npmjs.com/package/git-revision-webpack-plugin (4.0.0)

The latter one (which we currently use) has a bug (as described in pirelenito/git-revision-webpack-plugin#66). Therefore, we can't upgrade right now.

**Comment by tschmidtb51** [comment]
> The latter one (which we currently use) has a bug (as described in [pirelenito/git-revision-webpack-plugin#66](https://github.com/pirelenito/git-revision-webpack-plugin/issues/66)). Therefore, we can't upgrade right now. However, we could change to @hzzt/git-revision-webpack-plugin which I haven't checked yet.

There is an update available to 4.0.1 which fixes the compilation error:

```
[webpack-cli] Failed to load 'secvisogram/app/webpack.config.js' config
[webpack-cli] TypeError: GitRevisionPlugin is not a constructor
```

However, when opening `secvisogram/app/webpack.config.js` I still see the error message:

```
import GitRevisionPlugin
This expression is not constructable.
  Type 'typeof import("secvisogram/app/node_modules/git-revision-webpack-plugin/dist/index")' has no construct signatures.ts(2351)
```
To reproduce, see #44 .
